File: backend/src/main/java/com/ufc/backend/backend/exceptions/ClassesAndPreRequisiteAtTheSameTimeException.java
package com.ufc.backend.backend.exceptions;

import com.ufc.backend.backend.model.Classes;

public class ClassesAndPreRequisiteAtTheSameTimeException extends RuntimeException {
    public ClassesAndPreRequisiteAtTheSameTimeException(Classes classes) {
        super("Você não pode cursar: " + classes.getName() + " e " + classes.getPreRequisite().getName() + " no mesmo semestre");
    }

    public ClassesAndPreRequisiteAtTheSameTimeException(String msg, Throwable cause) {
        super(msg, cause);
    }
}

File: backend/src/main/java/com/ufc/backend/backend/exceptions/ClassesAndPreRequisiteAtTheSameTimeException.kt
package com.ufc.backend.backend.exceptions

import com.ufc.backend.backend.model.Subject

class ClassesAndPreRequisiteAtTheSameTimeException(subject: Subject) : RuntimeException(
    "Você não pode cursar: ${subject.name} e ${subject.preRequisite?.name} ao mesmo tempo"
)
